### ISSUE TITLE
Add more tests with corrupt videos

### ISSUE THREAD

**Issue by ahmadsharif1** [pytorch/torchcodec#165]
We should add tests with corrupt videos where there are missing frames, invalid header metadata, invalid packet headers, etc.

The goal is to ensure torchcodec detects this and returns the proper error message to the user.

In some cases like missing frames, it may be okay to keep going and handle it gracefully instead of returning EOFError.

**Comment by phuturesh** [comment]
Agree. Currently, the `VideoDecoder` may raise a RuntimeError or other exceptions when decoding videos that are partially corrupted. During the scanning process, it seems to just stop upon encountering the first error.

Ex. In a video containing 10,000 frames with corruption around the 4,000th frame, `VideoDecoder._num_frames` will only be about 4,000. The last 6,000 frames are discarded and we can not access them through `VideoDecoder`.

The thing is many real-world videos contain partial corruptions, yet various media players manage to ignore this and play them like there are no errors. Now the implementation of `VideoDecoder` may not properly align with this approach.

There may be a potential solution. In the scanning process we should somehow get the ``true'' num_frames. And in the decoding process, we can either

1. ignore a corrupted frame and return a 0 tensor.
2. or raise an Error, so programmers are to handle it. I prefer this one.

**Comment by NicolasHug** [comment]
I'll close this because this is difficult to *actively* work on that. We do test against corrupt videos, but more in a reactive way. It's close to impossible to generate *bad* videos, even with our VideoEncoder we have a hard time telling FFmpeg to do the wrong thing (it won't let us!)
